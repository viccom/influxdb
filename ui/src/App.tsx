import React, {SFC, ReactChildren} from 'react'

import RightClickLayer from 'src/clockface/components/right_click_menu/RightClickLayer'
import Nav from 'src/pageLayout'
import Notifications from 'src/shared/components/notifications/Notifications'
import LegendPortal from 'src/shared/components/LegendPortal'

interface Props {
  children: ReactChildren
}

const App: SFC<Props> = ({children}) => (
  <div className="chronograf-root">
    <Notifications />
    <RightClickLayer />
    <Nav />
    <LegendPortal />
    {children}
  </div>
)

export default App
