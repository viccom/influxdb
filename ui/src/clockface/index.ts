// Import Components
import AutoInput from './components/auto_input/AutoInput'
import Button from './components/Button'
import ConfirmationButton from './components/confirmation_button/ConfirmationButton'
import Dropdown, {DropdownMode} from './components/dropdowns/Dropdown'
import MultiSelectDropdown from './components/dropdowns/MultiSelectDropdown'
import Form from './components/form_layout/Form'
import Input, {InputType, AutoComplete} from './components/inputs/Input'
import MultipleInput, {
  MultiInputType,
} from './components/inputs/multipleInput/MultipleInput'
import OverlayTechnology from './components/overlays/OverlayTechnology'
import OverlayContainer from './components/overlays/OverlayContainer'
import OverlayHeading from './components/overlays/OverlayHeading'
import OverlayBody from './components/overlays/OverlayBody'
import OverlayFooter from './components/overlays/OverlayFooter'
import Panel from './components/panel/Panel'
import Radio from './components/radio_buttons/RadioButtons'
import SlideToggle from './components/slide_toggle/SlideToggle'
import WizardFullScreen from './components/wizard/WizardFullScreen'
import WizardOverlay from './components/wizard/WizardOverlay'
import WizardProgressHeader from './components/wizard/WizardProgressHeader'
import ProgressBar from './components/wizard/ProgressBar'
import ComponentSpacer from './components/component_spacer/ComponentSpacer'
import EmptyState from './components/empty_state/EmptyState'
import Spinner from './components/spinners/Spinner'
import SparkleSpinner from './components/spinners/SparkleSpinner'
import IndexList from './components/index_views/IndexList'
import Context from './components/context_menu/Context'
import FormElement from 'src/clockface/components/form_layout/FormElement'
import DraggableResizer from 'src/clockface/components/draggable_resizer/DraggableResizer'
import Label, {LabelType} from 'src/clockface/components/label/Label'
import LabelSelector from 'src/clockface/components/label/LabelSelector'
import GridSizer from 'src/clockface/components/grid_sizer/GridSizer'
import Select from 'src/clockface/components/Select'
import Grid from 'src/clockface/components/grid_layout/Grid'

// Import Types
import {
  ComponentColor,
  ComponentSize,
  ComponentStatus,
  DropdownMenuColors,
  DropdownChild,
  ButtonShape,
  Greys,
  IconFont,
  Columns,
  Alignment,
  Stack,
  ButtonType,
  Sort,
} from './types'

// Fire de lazer
export {
  Alignment,
  AutoComplete,
  AutoInput,
  Button,
  ButtonType,
  ButtonShape,
  Columns,
  ComponentColor,
  ComponentSize,
  ComponentSpacer,
  ComponentStatus,
  ConfirmationButton,
  Context,
  DraggableResizer,
  Dropdown,
  DropdownChild,
  DropdownMenuColors,
  DropdownMode,
  EmptyState,
  Form,
  FormElement,
  Greys,
  Grid,
  GridSizer,
  IconFont,
  IndexList,
  Input,
  InputType,
  Label,
  LabelSelector,
  LabelType,
  MultiSelectDropdown,
  MultiInputType,
  MultipleInput,
  OverlayBody,
  OverlayContainer,
  OverlayFooter,
  OverlayHeading,
  OverlayTechnology,
  Panel,
  ProgressBar,
  Radio,
  Select,
  SlideToggle,
  SparkleSpinner,
  Sort,
  Spinner,
  Stack,
  WizardFullScreen,
  WizardProgressHeader,
  WizardOverlay,
}
