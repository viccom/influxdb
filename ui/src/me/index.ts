import MePage from 'src/me/containers/MePage'
import Account from 'src/me/components/account/Account'

export {MePage, Account}
