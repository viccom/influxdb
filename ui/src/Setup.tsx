// Libraries
import React, {ReactElement, PureComponent} from 'react'
import {connect} from 'react-redux'
import {InjectedRouter} from 'react-router'

// APIs
import {getSetupStatus} from 'src/onboarding/apis'

// Actions
import {notify as notifyAction} from 'src/shared/actions/notifications'

// Components
import {ErrorHandling} from 'src/shared/decorators/errors'

// Utils
import {isOnboardingURL} from 'src/onboarding/utils'

// Types
import {Notification, NotificationFunc, RemoteDataState} from 'src/types'

interface State {
  loading: RemoteDataState
  isSetupComplete: boolean
}

interface Props {
  router: InjectedRouter
  children: ReactElement<any>
  notify: (message: Notification | NotificationFunc) => void
}

@ErrorHandling
export class Setup extends PureComponent<Props, State> {
  constructor(props: Props) {
    super(props)

    this.state = {
      loading: RemoteDataState.NotStarted,
      isSetupComplete: false,
    }
  }

  public async componentDidMount() {
    const {router} = this.props

    if (isOnboardingURL()) {
      this.setState({
        loading: RemoteDataState.Done,
      })
      return
    }

    const isSetupAllowed = await getSetupStatus()
    this.setState({
      loading: RemoteDataState.Done,
    })

    if (!isSetupAllowed) {
      return
    }

    router.push('/onboarding/0')
  }

  public render() {
    if (this.isLoading) {
      return <div className="page-spinner" />
    } else {
      return this.props.children && React.cloneElement(this.props.children)
    }
  }

  private get isLoading(): boolean {
    const {loading} = this.state
    return (
      loading === RemoteDataState.Loading ||
      loading === RemoteDataState.NotStarted
    )
  }
}

const mdtp = {
  notify: notifyAction,
}

export default connect(
  null,
  mdtp
)(Setup)
