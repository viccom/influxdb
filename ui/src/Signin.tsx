// Libraries
import React, {ReactElement, PureComponent} from 'react'
import {withRouter, WithRouterProps} from 'react-router'

// Components
import {ErrorHandling} from 'src/shared/decorators/errors'
import {getMe} from 'src/shared/apis/v2/user'

// Types
import {RemoteDataState} from 'src/types'

interface State {
  loading: RemoteDataState
}

interface OwnProps {
  children: ReactElement<any>
}

type Props = OwnProps & WithRouterProps

const FETCH_WAIT = 60000

@ErrorHandling
export class Signin extends PureComponent<Props, State> {
  private intervalID: NodeJS.Timer
  constructor(props: Props) {
    super(props)

    this.state = {
      loading: RemoteDataState.NotStarted,
    }
  }

  public async componentDidMount() {
    this.setState({loading: RemoteDataState.Done})
    this.checkForLogin()
    this.intervalID = setInterval(this.checkForLogin, FETCH_WAIT)
  }

  public componentWillUnmount() {
    clearInterval(this.intervalID)
  }

  public render() {
    if (this.isLoading) {
      return <div className="page-spinner" />
    }

    return this.props.children && React.cloneElement(this.props.children)
  }

  private get isLoading(): boolean {
    const {loading} = this.state
    return (
      loading === RemoteDataState.Loading ||
      loading === RemoteDataState.NotStarted
    )
  }

  private checkForLogin = async () => {
    try {
      await getMe()
    } catch (error) {
      clearInterval(this.intervalID)
      this.props.router.push('/signin')
    }
  }
}

export default withRouter(Signin)
