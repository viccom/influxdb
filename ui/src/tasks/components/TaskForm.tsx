// Libraries
import _ from 'lodash'
import React, {PureComponent, ChangeEvent} from 'react'

// Components
import {
  ComponentSpacer,
  Form,
  Grid,
  Columns,
  Input,
  Radio,
  ButtonShape,
  Button,
  ComponentColor,
  ButtonType,
} from 'src/clockface'
import TaskOptionsOrgDropdown from 'src/tasks/components/TasksOptionsOrgDropdown'
import TaskScheduleFormField from 'src/tasks/components/TaskScheduleFormField'

// Types
import {TaskOptions, TaskSchedule} from 'src/utils/taskOptionsToFluxScript'
import {Alignment, Stack, ComponentStatus} from 'src/clockface/types'
import {Organization} from 'src/api'

// Styles
import './TaskForm.scss'

interface Props {
  orgs: Organization[]
  taskOptions: TaskOptions
  onChangeScheduleType: (schedule: TaskSchedule) => void
  onChangeInput: (e: ChangeEvent<HTMLInputElement>) => void
  onChangeTaskOrgID: (orgID: string) => void
  isInOverlay?: boolean
  onSubmit?: () => void
  canSubmit?: boolean
  dismiss?: () => void
}

interface State {
  schedule: TaskSchedule
}

export default class TaskForm extends PureComponent<Props, State> {
  public static defaultProps: Partial<Props> = {
    isInOverlay: false,
    onSubmit: () => {},
    canSubmit: true,
    dismiss: () => {},
  }
  constructor(props) {
    super(props)

    this.state = {
      schedule: props.taskOptions.taskScheduleType,
    }
  }

  public render() {
    const {
      onChangeInput,
      onChangeTaskOrgID,
      taskOptions: {name, taskScheduleType, interval, offset, cron, orgID},
      orgs,
      isInOverlay,
    } = this.props

    return (
      <Form>
        <Grid>
          <Grid.Row>
            <Grid.Column widthXS={Columns.Twelve}>
              <Form.Element label="Name">
                <Input
                  name="name"
                  placeholder="Name your task"
                  onChange={onChangeInput}
                  value={name}
                />
              </Form.Element>
            </Grid.Column>
            <Grid.Column widthXS={Columns.Twelve}>
              <Form.Element label="Owner">
                <TaskOptionsOrgDropdown
                  orgs={orgs}
                  selectedOrgID={orgID}
                  onChangeTaskOrgID={onChangeTaskOrgID}
                />
              </Form.Element>
            </Grid.Column>
            <Grid.Column>
              <Form.Element label="Schedule Task">
                <ComponentSpacer
                  align={Alignment.Left}
                  stackChildren={Stack.Rows}
                >
                  <Radio shape={ButtonShape.StretchToFit}>
                    <Radio.Button
                      id="interval"
                      active={taskScheduleType === TaskSchedule.interval}
                      value={TaskSchedule.interval}
                      titleText="Interval"
                      onClick={this.handleChangeScheduleType}
                    >
                      Interval
                    </Radio.Button>
                    <Radio.Button
                      id="cron"
                      active={taskScheduleType === TaskSchedule.cron}
                      value={TaskSchedule.cron}
                      titleText="Cron"
                      onClick={this.handleChangeScheduleType}
                    >
                      Cron
                    </Radio.Button>
                  </Radio>
                </ComponentSpacer>
              </Form.Element>
            </Grid.Column>
            <TaskScheduleFormField
              onChangeInput={onChangeInput}
              schedule={taskScheduleType}
              interval={interval}
              offset={offset}
              cron={cron}
            />
            {isInOverlay && this.buttons}
          </Grid.Row>
        </Grid>
      </Form>
    )
  }

  private get buttons(): JSX.Element {
    const {onSubmit, canSubmit, dismiss} = this.props
    return (
      <Grid.Column widthXS={Columns.Twelve}>
        <Form.Footer>
          <Button
            text="Cancel"
            onClick={dismiss}
            titleText="Cancel save"
            type={ButtonType.Button}
          />
          <Button
            text={'Save as Task'}
            color={ComponentColor.Success}
            type={ButtonType.Submit}
            onClick={onSubmit}
            status={
              canSubmit ? ComponentStatus.Default : ComponentStatus.Disabled
            }
          />
        </Form.Footer>
      </Grid.Column>
    )
  }

  private handleChangeScheduleType = (schedule: TaskSchedule): void => {
    this.props.onChangeScheduleType(schedule)
  }
}
