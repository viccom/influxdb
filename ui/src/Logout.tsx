// Libraries
import {PureComponent} from 'react'
import {withRouter, WithRouterProps} from 'react-router'

// APIs
import {logout} from 'src/me/apis'

// Components
import {ErrorHandling} from 'src/shared/decorators/errors'

type Props = WithRouterProps

@ErrorHandling
export class Logout extends PureComponent<Props> {
  public componentDidMount() {
    this.handleSignOut()
  }

  public render() {
    return null
  }

  private handleSignOut = async () => {
    await logout()
    this.props.router.push('/')
  }
}

export default withRouter<Props>(Logout)
