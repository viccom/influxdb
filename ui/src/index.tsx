import 'babel-polyfill'

import React, {PureComponent} from 'react'
import {render} from 'react-dom'
import {Provider} from 'react-redux'
import {Router, Route, useRouterHistory, IndexRoute} from 'react-router'
import {createHistory, History} from 'history'

import configureStore from 'src/store/configureStore'
import {loadLocalStorage} from 'src/localStorage'

import {getRootNode} from 'src/utils/nodes'
import {getBasepath} from 'src/utils/basepath'

// Components
import App from 'src/App'
import GetSources from 'src/shared/containers/GetSources'
import SetActiveSource from 'src/shared/containers/SetActiveSource'
import GetOrganizations from 'src/shared/containers/GetOrganizations'
import Setup from 'src/Setup'
import Signin from 'src/Signin'
import SigninPage from 'src/onboarding/containers/SigninPage'
import Logout from 'src/Logout'
import TaskPage from 'src/tasks/containers/TaskPage'
import TasksPage from 'src/tasks/containers/TasksPage'
import OrganizationsIndex from 'src/organizations/containers/OrganizationsIndex'
import OrganizationView from 'src/organizations/containers/OrganizationView'
import TaskEditPage from 'src/tasks/containers/TaskEditPage'
import DashboardPage from 'src/dashboards/components/DashboardPage'
import DashboardsIndex from 'src/dashboards/components/dashboard_index/DashboardsIndex'
import DataExplorerPage from 'src/dataExplorer/components/DataExplorerPage'
import {MePage, Account} from 'src/me'
import NotFound from 'src/shared/components/NotFound'
import GetLinks from 'src/shared/containers/GetLinks'
import GetMe from 'src/shared/containers/GetMe'
import SourcesPage from 'src/sources/components/SourcesPage'
import ConfigurationPage from 'src/configuration/components/ConfigurationPage'

import OnboardingWizardPage from 'src/onboarding/containers/OnboardingWizardPage'

// Actions
import {disablePresentationMode} from 'src/shared/actions/app'

// Styles
import 'src/style/chronograf.scss'

const rootNode = getRootNode()
const basepath = getBasepath()

declare global {
  interface Window {
    basepath: string
  }
}

// Older method used for pre-IE 11 compatibility
window.basepath = basepath

const history: History = useRouterHistory(createHistory)({
  basename: basepath, // this is written in when available by the URL prefixer middleware
})

const store = configureStore(loadLocalStorage(), history)
const {dispatch} = store

history.listen(() => {
  dispatch(disablePresentationMode())
})

window.addEventListener('keyup', event => {
  const escapeKeyCode = 27
  // fallback for browsers that don't support event.key
  if (event.key === 'Escape' || event.keyCode === escapeKeyCode) {
    dispatch(disablePresentationMode())
  }
})

class Root extends PureComponent {
  public render() {
    return (
      <Provider store={store}>
        <Router history={history}>
          <Route component={GetLinks}>
            <Route component={Setup}>
              <Route path="/onboarding">
                <Route path=":stepID" component={OnboardingWizardPage} />
                <Route
                  path=":stepID/:substepID"
                  component={OnboardingWizardPage}
                />
              </Route>
              <Route path="/signin" component={SigninPage} />
              <Route path="/logout" component={Logout} />
              <Route path="/">
                <Route component={Signin}>
                  <Route component={GetMe}>
                    <Route component={GetOrganizations}>
                      <Route component={App}>
                        <Route component={GetSources}>
                          <Route component={SetActiveSource}>
                            <IndexRoute component={DashboardsIndex} />
                            <Route path="organizations">
                              <IndexRoute component={OrganizationsIndex} />
                              <Route
                                path=":orgID/:tab"
                                component={OrganizationView}
                              />
                            </Route>
                            <Route path="tasks">
                              <IndexRoute component={TasksPage} />
                              <Route path="new" component={TaskPage} />
                              <Route path=":id" component={TaskEditPage} />
                            </Route>
                            <Route
                              path="data-explorer"
                              component={DataExplorerPage}
                            />
                            <Route path="dashboards">
                              <IndexRoute component={DashboardsIndex} />
                              <Route
                                path=":dashboardID"
                                component={DashboardPage}
                              />
                            </Route>
                            <Route path="me" component={MePage} />
                            <Route path="account/:tab" component={Account} />
                            <Route path="sources" component={SourcesPage} />
                            <Route
                              path="configuration/:tab"
                              component={ConfigurationPage}
                            />
                          </Route>
                        </Route>
                      </Route>
                    </Route>
                  </Route>
                </Route>
              </Route>
            </Route>
          </Route>
          <Route path="*" component={NotFound} />
        </Router>
      </Provider>
    )
  }
}

if (rootNode) {
  render(<Root />, rootNode)
}
