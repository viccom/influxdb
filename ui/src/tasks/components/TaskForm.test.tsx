// Libraries
import React from 'react'
import {shallow} from 'enzyme'

// Components
import TaskForm from 'src/tasks/components/TaskForm'
import {Form} from 'src/clockface'

// Constants
import defaultTaskOptions from 'src/tasks/reducers/v2'

const setup = (override?) => {
  const props = {
    orgs: [],
    taskOptions: defaultTaskOptions,
    onChangeScheduleType: jest.fn(),
    onChangeInput: jest.fn(),
    onChangeTaskOrgID: jest.fn(),
    ...override,
  }

  const wrapper = shallow(<TaskForm {...props} />)

  return {wrapper}
}

describe('TaskForm', () => {
  describe('rendering', () => {
    it('renders', () => {
      const {wrapper} = setup()
      const form = wrapper.find(Form)

      expect(wrapper.exists()).toBe(true)
      expect(form.exists()).toBe(true)
      expect(wrapper).toMatchSnapshot()
    })
  })
})
