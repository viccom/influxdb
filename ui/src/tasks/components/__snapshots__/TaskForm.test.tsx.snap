// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskForm rendering renders 1`] = `
<Form>
  <Grid>
    <GridRow>
      <GridColumn
        widthXS={12}
      >
        <FormElement
          label="Name"
        >
          <Input
            autoFocus={false}
            autocomplete="off"
            disabledTitleText="This input is disabled"
            name="name"
            onChange={[MockFunction]}
            placeholder="Name your task"
            size="sm"
            spellCheck={false}
            status="default"
            titleText=""
            value=""
          />
        </FormElement>
      </GridColumn>
      <GridColumn
        widthXS={12}
      >
        <FormElement
          label="Owner"
        >
          <TaskOptionsOrgDropdown
            onChangeTaskOrgID={[MockFunction]}
            orgs={Array []}
          />
        </FormElement>
      </GridColumn>
      <GridColumn
        widthXS={12}
      >
        <FormElement
          label="Schedule Task"
        >
          <ComponentSpacer
            align="left"
            stackChildren="rows"
          >
            <Radio
              color="primary"
              shape="stretch"
              size="sm"
            >
              <RadioButton
                active={false}
                disabled={false}
                disabledTitleText="This option is disabled"
                id="interval"
                onClick={[Function]}
                titleText="Interval"
                value="interval"
              >
                Interval
              </RadioButton>
              <RadioButton
                active={false}
                disabled={false}
                disabledTitleText="This option is disabled"
                id="cron"
                onClick={[Function]}
                titleText="Cron"
                value="cron"
              >
                Cron
              </RadioButton>
            </Radio>
          </ComponentSpacer>
        </FormElement>
      </GridColumn>
      <TaskScheduleFormFields
        onChangeInput={[MockFunction]}
      />
    </GridRow>
  </Grid>
</Form>
`;
